mlflow/server/js/src/components/EditableTagsTableView.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Utils from '../utils/Utils';
+import PropTypes from 'prop-types';
+import { Form, Input, Button, message } from 'antd';
+import { getUUID, setTagApi } from '../Actions';
+import { EditableFormTable } from './tables/EditableFormTable';
+import _ from 'lodash';
+
+export class EditableTagsTableView extends React.Component {
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    tags: PropTypes.object.isRequired,
+    form: PropTypes.object.isRequired,
+    setTagApi: PropTypes.func.isRequired,
+  };
+
+  state = { isRequestPending: false };
+
+  tableColumns = [
+    {
+      title: 'Name',
+      dataIndex: 'name',
+      width: 200,
+    },
+    {
+      title: 'Value',
+      dataIndex: 'value',
+      width: 200,
+      editable: true,
+    }
+  ];
+
+  requestId = getUUID();
+
+  getData = () => _.sortBy(Utils.getVisibleTagValues(this.props.tags).map((values) => ({
+    key: values[0],
+    name: values[0],
+    value: values[1],
+  })), 'name');
+
+  getTagNamesAsSet = () => new Set(
+    Utils.getVisibleTagValues(this.props.tags).map((values) => values[0])
+  );
+
+  handleAddTag = (e) => {
+    e.preventDefault();
+    const { form, runUuid, setTagApi: setTag } = this.props;
+    form.validateFields((err, values) => {
+      if (!err) {
+        this.setState({ isRequestPending: true });
+        setTag(runUuid, values.name, values.value, this.requestId)
+          .then(() => {
+            this.setState({ isRequestPending: false });
+            form.resetFields();
+          })
+          .catch((ex) => {
+            this.setState({ isRequestPending: false });
+            console.error(ex);
+            message.error('Failed to add tag.');
```

Comment:
Should we consider displaying a user-visible error message in this (hopefully rare) case to help with debugging? We could do so via e.g. `message.error('Failed to add tag. Error: ' + ex.getUserVisibleError())` - we currently do this in [NoteEditorView](https://github.com/mlflow/mlflow/blob/a089618bfa509e71f8dfaffc0f42f13e71aa7942/mlflow/server/js/src/components/NoteEditorView.js#L90) (for adding notes) and [RenameRunModal](https://github.com/mlflow/mlflow/blob/master/mlflow/server/js/src/components/modals/RenameRunModal.js#L71) (when renaming a run), it seems helpful IMO.

Reply:
Changed accordingly.